File: scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/project/bsp/data/BspBundle.scala
package org.jetbrains.plugins.scala.project.bsp.data

import com.intellij.DynamicBundle
import org.jetbrains.annotations.Nls
import org.jetbrains.annotations.PropertyKey

object BspBundle {
  private final val BUNDLE = "messages.ScalaBspBundle"
  private final val INSTANCE = new BspBundle

  @Nls def message(@PropertyKey(resourceBundle = BUNDLE) key: String, params: AnyRef*): String = INSTANCE.getMessage(key, params)
}
final class BspBundle private extends DynamicBundle(BspBundle.BUNDLE) {
}

File: bsp-builtin/bsp/src/org/jetbrains/bsp/BspBundle.java
package org.jetbrains.bsp;

import com.intellij.DynamicBundle;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.PropertyKey;

public final class BspBundle extends DynamicBundle {
    @NonNls
    private static final String BUNDLE = "messages.ScalaBspBundle";

    private static final BspBundle INSTANCE = new BspBundle();

    private BspBundle() {
        super(BUNDLE);
    }

    @Nls
    public static String message(@NotNull @PropertyKey(resourceBundle = BUNDLE) String key, @NotNull Object... params) {
        return INSTANCE.getMessage(key, params);
    }
}
